### ISSUE TITLE
[SIP] It is proposed to add the feature of grouping like folders

### ISSUE THREAD

**Issue by poweronlevel** [apache/superset#29236]
It is proposed to add the feature of grouping similar to folders, because when there are dozens or even hundreds of lists under dashborads, charts, datasets or Saved SQLs, it is difficult to find them, although there is a search function.

**Comment by ttdpro98** [comment]
True, we really need it, like the folder in Grafana. And we can set permissions for each folder.

**Comment by rusackas** [comment]
Hi there. Thanks for asking for this feature, and rest assured you're not the first. 

If you want to organize and/or contribute this feature, we'd like to see the remainder of the SIP template filled out. This is to answer questions about how it will be built, what it will look like, what are the alternatives (e.g. what does the Tags feature _not_ accomplish for your needs), etc. 

If this is just a feature request, we can move/merge this to a GitHub Discussion under the "Ideas" category. If you want to execute on this idea, we just need more details so we can discuss it properly.

**Comment by geido** [comment]
Converting this to a discussion as it does not follow the SIP format. Please, feel free to re-open using the SIP guidelines.
